services/forest/src/service/SCIHF.js

```diff
@@ -0,0 +1,150 @@
+        };
+      });
+    },
+
+    /**
+     * Get the layer of the forest structural condition index crossed with human footprint
+     * for a given area
+     *
+     * @param {String | Number} areaType area type
+     * @param {String} areaId area id
+     *
+     * @returns {Object} Geojson object with the geometry
+     */
+    getSCIHFLayer: async (areaType, areaId) => {
+      let data;
+      switch (areaType) {
+        case 'ea':
+          data = await SCIHFPersistence.findSCIHFLayerInEA(areaId);
+          break;
+        case 'states':
+          data = await SCIHFPersistence.findSCIHFLayerInState(areaId);
+          break;
+        case 'basinSubzone':
+          data = await SCIHFPersistence.findSCIHFLayerInBasinSubzone(areaId);
+          break;
+        case 'pa': {
+          const binaryProtected = await restAPI.requestBinaryProtectedByCategory(areaId);
+          data = await SCIHFPersistence.findSCIHFLayerInPA(binaryProtected.binary_protected);
+          break;
+        }
+        default:
+          data = null;
+          break;
+      }
+
+      if (!data) {
+        throw new Error('Data layer for SCIHF doesn\'t exists to the selected area id and area type');
+      }
+
+      return data;
+    },
+
+    /**
+     * Get the layer of one combination of forest structural condition index category and a human
+     * footprint persistence category, divided by protected areas for a given area
+     *
+     * @param {String} sciCat sci category
+     * @param {String} hfPers human footprint persistence category
+     * @param {String | Number} areaType area type
+     * @param {String} areaId area id
+     *
+     * @returns {Object} Geojson object with the geometry
+     */
+    getSCIHFPALayer: async (sciCat, hfPers, areaType, areaId) => {
+      let data;
+      switch (areaType) {
+        case 'ea':
+          data = await SCIHFPersistence.findSCIHFPALayerInEA(areaId, sciCat, hfPers);
+          break;
+        case 'states':
+          data = await SCIHFPersistence.findSCIHFPALayerInState(areaId, sciCat, hfPers);
+          break;
+        case 'basinSubzone':
+          data = await SCIHFPersistence.findSCIHFPALayerInBasinSubzone(areaId, sciCat, hfPers);
+          break;
+        case 'pa': {
+          const binaryProtected = await restAPI.requestBinaryProtectedByCategory(areaId);
+          data = await SCIHFPersistence.findSCIHFPALayerInPA(
+            binaryProtected.binary_protected,
+            sciCat,
+            hfPers,
+          );
+          break;
+        }
+        default:
+          data = null;
+          break;
+      }
+
+      if (!data) {
+        throw new Error('Data layer for SCIHF PA doesn\'t exists to the selected area id and area type');
+      }
+
+      const binaryProtectedValues = [...new Set(data.features.map(b => (b.properties.binary_protected)))].join('&');
+      const categoriesPA = await restAPI.requestCategoriesByBinaryProtected(binaryProtectedValues);
```

Comment:
Control errors from this function (same as the first function)

Reply:
Done